bookwyrm/static/js/bookwyrm.js

```diff
@@ -430,4 +433,38 @@ let BookWyrm = new class {
         parent.appendChild(label)
         parent.appendChild(input)
     }
+
+    /**
+     * Set up a "click-to-copy" component from a textarea element
+     * with `data-copytext`, `data-copytext-label`, `data-copytext-success`
+     * attributes.
+     *
+     * @param  {object}  node - DOM node of the text container
+     * @return {undefined}
+     */
+
+    copyText(textareaEl) {
+        const text = textareaEl.textContent;
+
+        const copyButtonEl = document.createElement('button');
+
+        copyButtonEl.textContent = textareaEl.dataset.copytextLabel;
+        copyButtonEl.classList.add(
+            "mt-2",
+            "button",
+            "is-small",
+            "is-fullwidth",
+            "is-primary",
+            "is-light"
+        );
```

Comment:
why generate this element in javascript as opposed to putting it in the template?

Reply:
If the client doesn't use JS, the button in the template will do nothing. In that case I prefer showing only a textarea in which the user can select and copy manually and no expectation of interactivity with a button. If the client uses JS, it can add the button with interactivity.